### PR TITLE
New package: Orthanc-1.11.2

### PR DESCRIPTION
**Pull request by gcarlos64**
#### Testing the changes
- I tested the changes in this PR: **briefly**

#### New package
- This new package conforms to the [quality requirements](https://github.com/void-linux/void-packages/blob/master/Manual.md#quality-requirements): **YES**

#### Local build testing
- I built this PR locally for my native architecture, (x86_64-LIBC)

### CODE REVIEW

**Comment by abenson** [comment]
One commit per template

File: common/shlibs [hunk 1]
```diff
@@ -4185,3 +4185,41 @@ libnvidia-container-go.so.1 libnvidia-container-1.10.0_1
 libqtspell-qt5.so.1 qtspell-1.0.1_1
 libopensmtpd.so.0 libopensmtpd-0.7_1
 libiio.so.0 libiio-0.23_1
+libcivetweb.so.1 libcivetweb-1.15_1
+libcmr.so.17
```
Comment (Chocimier) [inline common/shlibs:4189]: typo

File: common/shlibs [hunk 2]
```diff
@@ -4185,3 +4185,41 @@ libnvidia-container-go.so.1 libnvidia-container-1.10.0_1
...
+libdcmjpls.so.17 libdcmtk-3.6.7_1
+libijg16.so.17 libdcmtk-3.6.7_1
+libdcmnet.so.17 libdcmtk-3.6.7_1
+libijg8.so.17 libdcmtk-3.6.7_1
+libdcmpmap.so.17 libdcmtk-3.6.7_1
+liboflog.so.17 libdcmtk-3.6.7_1
+libdcmpstat.so.17 libdcmtk-3.6.7_1
+libofstd.so.17 libdcmtk-3.6.7_1
+libdcmqrdb.so.17 libdcmtk-3.6.7_1
+libDelayedDeletion.so.1.11.1 Orthanc-plugins-1.11.1_1
```
Comment (Chocimier) [inline common/shlibs:4217]: common/shlibs is meant to only contain solibs used by otehr packages - are plugins like that?

File: srcpkgs/Orthanc-DicomWeb/template [hunk 3]
```diff
@@ -0,0 +1,22 @@
+# Template file for 'Orthanc-DicomWeb'
+pkgname=Orthanc-DicomWeb
+version=1.9
+revision=1
+wrksrc="${pkgname/-}-${version}"
+build_style=cmake
+configure_args="-DALLOW_DOWNLOADS=ON -DORTHANC_FRAMEWORK_WOURCE=path
+ -DORTHANC_FRAMEWORK_ROOT=/usr/share/orthanc/OrthancFramework/Sources"
+hostmakedepends="tar unzip python3"
+makedepends="Orthanc-devel>=1.11.1 pugixml-devel gtest-devel jsoncpp-devel
```
Comment (Chocimier) [inline srcpkgs/Orthanc-DicomWeb/template:10]: Drop patterns from build deps, they break build and are not restricting anyway.

File: srcpkgs/Orthanc/files/orthanc/run [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+then
+	mkdir -pm 0750 $ORTHANCSTORAGE &&
+	chown ${USER}:${USER} $ORTHANCSTORAGE ||
+	exit 1
+fi
+
+# Enable log
+if [ -n $LOG ]; then
+	: ${LOGDIR:="/log/orthanc"}
+	OPTS="${OPTS}--logdir=$LOGDIR "
```
Comment (ahesford) [inline srcpkgs/Orthanc/files/orthanc/run:21]: This weird trailing space behavior is likely to fail if somebody modifies `$OPTS` in `conf`. Better to do
```sh
OPTS="${OPTS} --logdir=$LOGDIR"
```
Even if `$OPTS` is empty, there's no harm in a leading space.

File: srcpkgs/Orthanc/files/orthanc/run [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+	if [ ! -e $LOGDIR ]; then
+		mkdir -pm 0750 $LOGDIR &&
+		chown ${USER}:${USER} $LOGDIR ||
+		exit 1
+	fi
+fi
+
+# Set OPTS on conf file for aditional parameters
+OPTS="$OPTS$CONFIGFILE"
+exec chpst -u ${USER}:${USER} Orthanc $OPTS
```
Comment (ahesford) [inline srcpkgs/Orthanc/files/orthanc/run:32]: Just do
```suggestion
exec chpst -u ${USER}:${USER} Orthanc ${OPTS} "${CONFIGFILE}"
```

File: srcpkgs/Orthanc/files/Configuration.json [hunk 6]
```diff
@@ -0,0 +1,915 @@
+{
```
Comment (Chocimier) [inline srcpkgs/Orthanc/files/Configuration.json:1]: adding to distfiles will prevent this from going out of sync

File: srcpkgs/Orthanc/files/orthanc/run [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+OPTS=""
+[ -r conf ] && . ./conf
+
+: ${USER:="_orthanc"}
+
+: ${CONFIGFILE:="/etc/orthanc/Configuration.json"}
+[ -r $CONFIGFILE ] || exit 1
+
+: ${ORTHANCSTORAGE:="/var/lib/OrthancStorage"}
+if [ ! -e $ORTHANCSTORAGE ]
```
Comment (ahesford) [inline srcpkgs/Orthanc/files/orthanc/run:11]: You should be quoting your use of these variables. Users might modify the `conf` file and specify spaces in the paths.

**Review by ahesford — CHANGES_REQUESTED** [review]
General comments (this is not exhaustive):
- Are all of these shared libs really necessary for linking by other packages? If not, leave them out of `common/shlibs`.
- One commit per package, one package per commit. Split each of these into a separate commit in the same PR.
- Remove version restrictions in `hostmakedepends` and `makedepends`. Version enforcement makes no sense here because the repository specifies a single version of the package; whatever is current. It either builds with that version or it's broken.
-  What is the purpose of the `Orthanc-{DicomWeb,PostgreSQL,Python}` packages? They aren't reference by any other templates.
    - If `Orthanc-Python` really doesn't work with py3.11, that's a problem, because we'll be moving to py3.11 soon and I'd like to minimize potential breakage caused by the addition of new specialty packages in the meantime.
    - If `Orthanc-Python` is a Python package, it should depend on `python3`.

File: srcpkgs/dcmtk/template [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+revision=1
+build_style=cmake
+configure_args="-DDCMTK_ENABLE_CXX11=ON -DBUILD_SHARED_LIBS=ON -DDCMTK_WITH_DOXYGEN=ON"
+hostmakedepends="doxygen"
+makedepends="openssl-devel zlib-devel tiff-devel libpng-devel libxml2-devel icu-devel"
+short_desc="Libraries and tools that implements DICOM standard"
+maintainer="Carlos E. Gallo F. <gcarlos@disroot.org>"
+license="custom: see /usr/share/licenses/${pkgname}"
+homepage="https://dcmtk.org/dcmtk.php.en"
+distfiles="https://dicom.offis.de/download/dcmtk/dcmtk367/dcmtk-${version}.tar.gz"
```
Comment (Chocimier) [inline srcpkgs/dcmtk/template:13]: ${version//./}

File: common/shlibs (hunk 2 above)
Comment (bingulo) [reply common/shlibs]: these plugins will probably never be used by other pkgs, I removed its from shlibs

**Comment by bingulo** [comment]
@ahesford
>     * Are all of these shared libs really necessary for linking by other packages? If not, leave them out of `common/shlibs`.

Done.

>     * One commit per package, one package per commit. Split each of these into a separate commit in the same PR.

Done.

>     * Remove version restrictions in `hostmakedepends` and `makedepends`. Version enforcement makes no sense here because the repository specifies a single version of the package; whatever is current. It either builds with that version or it's broken.

Done.

>     * What is the purpose of the `Orthanc-{DicomWeb,PostgreSQL,Python}` packages? They aren't reference by any other templates.

These are some plugins i decided to package, it isn't requiered by the main package (Orthanc), but as these makedepends of Orthanc I added them into this PR.
      
>       * If `Orthanc-Python` really doesn't work with py3.11, that's a problem, because we'll be moving to py3.11 soon and I'd like to minimize potential breakage caused by the addition of new specialty packages in the meantime.

It works, you just need to pass a flag with the major python version. As I passed python 3.10, i decided to restrict it on the template, but I understand now that it doesn't make sense and I've removed it. Now I need to get the python version from the upstream repo and pass it to configure_args. I'll try some way to do it.

>       * If `Orthanc-Python` is a Python package, it should depend on `python3`.

It isn't, it's mostly like a binding instead.

**Comment by Chocimier** [comment]
> I need to get the python version from the upstream repo and pass it to configure_args 

see https://github.com/void-linux/void-packages/blob/master/Manual.md#python-packages

File: srcpkgs/Orthanc/files/orthanc/run (hunk 4 above)
Comment (bingulo) [reply srcpkgs/Orthanc/files/orthanc/run]: I replaced to `OPTS="${OPTS} --logdir=\"$LOGDIR\""` instead.

File: srcpkgs/Orthanc/files/orthanc/run (hunk 5 above)
Comment (bingulo) [reply srcpkgs/Orthanc/files/orthanc/run]: Done

File: srcpkgs/Orthanc/files/Configuration.json (hunk 6 above)
Comment (bingulo) [reply srcpkgs/Orthanc/files/Configuration.json:1]: It's already distributed on distfile, I just changed some default settings (specifically to match $ORTHANCSTORAGE on service). I think I'll run a sed on it instead.

Comment (classabbyamp) [reply srcpkgs/Orthanc/files/Configuration.json:1]: a patch would probably be better

Comment (bingulo) [reply srcpkgs/Orthanc/files/Configuration.json]: @classabbyamp I done it

**Comment by bingulo** [comment]
> > I need to get the python version from the upstream repo and pass it to configure_args
> 
> see https://github.com/void-linux/void-packages/blob/master/Manual.md#python-packages

thx, now I'm using the py3_ver var.

File: srcpkgs/Orthanc-DicomWeb/template [hunk 9]
```diff
@@ -0,0 +1,22 @@
+# Template file for 'Orthanc-DicomWeb'
+pkgname=Orthanc-DicomWeb
+version=1.9
+revision=1
+wrksrc="${pkgname/-}-${version}"
+build_style=cmake
+configure_args="-DALLOW_DOWNLOADS=ON -DORTHANC_FRAMEWORK_WOURCE=path
```
Comment (Chocimier) [inline srcpkgs/Orthanc-DicomWeb/template:7]: typo across templates?

File: srcpkgs/Orthanc/template [hunk 10]
```diff
@@ -0,0 +1,58 @@
...
+ sqlite-devel libcivetweb-devel openssl-devel jsoncpp-devel lua-devel
+ pugixml-devel"
+depends="dcmtk"
+short_desc="Lightweight RESTful DICOM server"
+maintainer="Carlos E. Gallo F. <gcarlos@disroot.org>"
+license="GPL-3.0-or-later"
+homepage="https://www.orthanc-server.com/index.php"
+distfiles="https://www.orthanc-server.com/downloads/get.php?path=/orthanc/Orthanc-${version}.tar.gz"
+checksum=a634adc9f6dd0a74ebaeb415657b5f3d51822b4fb347641dc44f3bf8b5b56db5
+ignore_elf_dirs="/usr/share/orthanc"
```
Comment (Chocimier) [inline srcpkgs/Orthanc/template:20]: ignore_elf_dirs is for emulator artifacts and such. Actual executable code like here need to be moved elsewhere, e.g. /usr/lib/orthanc

File: srcpkgs/dcmtk/template [hunk 11]
```diff
@@ -0,0 +1,49 @@
...
+revision=1
+build_style=cmake
+configure_args="-DDCMTK_ENABLE_CXX11=ON -DBUILD_SHARED_LIBS=ON -DDCMTK_WITH_DOXYGEN=ON"
+hostmakedepends="doxygen"
+makedepends="openssl-devel zlib-devel tiff-devel libpng-devel libxml2-devel icu-devel"
+short_desc="Libraries and tools that implements DICOM standard"
+maintainer="Carlos E. Gallo F. <gcarlos@disroot.org>"
+license="custom: see /usr/share/licenses/${pkgname}"
+homepage="https://dcmtk.org/dcmtk.php.en"
+distfiles="https://dicom.offis.de/download/dcmtk/${version//.}/dcmtk-${version}.tar.gz"
```
Comment (Chocimier) [inline srcpkgs/dcmtk/template:13]: typo, does not fetch

**Review by Chocimier — COMMENTED** [review]
Please check that it still builds after a round of review.

File: srcpkgs/Orthanc-DicomWeb/template (hunk 9 above)
Comment (bingulo) [reply srcpkgs/Orthanc-DicomWeb/template:7]: yes, I will fix it

File: srcpkgs/Orthanc/template (hunk 10 above)
Comment (bingulo) [reply srcpkgs/Orthanc/template:20]: okay, I will move it to /usr/l/ib/orthanc/plugins/ dir

**Comment by bingulo** [comment]
Should I specify python3 on depends of Orthanc-Python? It don't explicity depends, but if it's compiled with python 3.10, as example, you cannot use it with python 3.11, as you need to compile with the specific python version flag. I'm thinking in put something like `depends="python3>=${py3_ver}<$(( ${py3_ver} + 1 ))"` (probably it shell expansion don't work, it's just to illustrate my idea)

**Comment by ahesford** [comment]
No, just depend on `python3`. We only support one version of py3 in the repository and I bump *every* Python package when we do a minor version jump.

**Comment by bingulo** [comment]
> No, just depend on `python3`. We only support one version of py3 in the repository and I bump _every_ Python package when we do a minor version jump.

Of course, but I'm thinking on some case as version hold by user. In that example, the xbps shouldn't allow to upgrade python to 3.11 if the user holds Orthanc-Python on a version wich depends on 3.10.

**Comment by bingulo** [comment]
Now all the packages are building successfully on my local.

I'll try to cross build to the others archs now.

**Comment by Chocimier** [comment]
> version hold by user

This is already covered by dynamic linking tracking, look for `SONAME: libpython3.10.so.1.0 <-> python3>=3.10.0_1` in build log.

**Comment by bingulo** [comment]
> > version hold by user
> 
> This is already covered by dynamic linking tracking, look for `SONAME: libpython3.10.so.1.0 <-> python3>=3.10.0_1` in build log.

Ahh okay, so I don't need to specify python3 on depends?

**Comment by bingulo** [comment]
I marked all packages (except civetweb) since dcmtk, a dependency of Orthanc, need running some code on target arch to generate a header for crossbuild. It can be done running a compiled test (a single .cc present on distfile) on each arch and providing its generated headers together to package and copying it to desired directory, what seems a little tricky to maintain.

**Comment by anhangah** [comment]
I just updated Orthanc and Orthanc-DicomWeb to upstream version and made a change o Orthanc template accordingly Orthanc documentation. Also I tested the build on native aarch64 and everything seems fine.

**Pull request closed without merging**